### PR TITLE
C++ Example setup, examples for point3d/arrow3d/disconnectedspace

### PR DESCRIPTION
**Pull request by Wumpf**
### What

Adds C++ example build setup and examples for:

* Point3D  #2789
* Arrow3D #2785
* DisconnectedSpace #2791

as before Point2D example is blocked on rects for the moment.

Generating a bunch more methods now for better usability. Still lacking user extensions though, in particular Point3D construction is poor right now

Examples can be built in bulk with ` ./tests/cpp/build_all_doc_examples.sh`
Then run with `./build/tests/cpp/doc_example_point3d_EXAMPLE_NAME`
They all connect to localhost on default port for now.

Point3D Simple:
image

Point3D Random:
image

Arrow3D:
image

Disconnected Space:
image

Dependent on #2906 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/2908) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/2908)
- [Docs preview](https://rerun.io/preview/pr%3Aandreas%2Fcpp%2Fexamplesv2/docs)
- [Examples preview](https://rerun.io/preview/pr%3Aandreas%2Fcpp%2Fexamplesv2/examples)

### CODE REVIEW

File: docs/code-examples/point3d_random_v2.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    auto rr_stream = rr::RecordingStream("points3d_random");
+    rr_stream.connect("127.0.0.1:9876");
+
+    std::default_random_engine gen;
+    std::uniform_real_distribution<float> dist_pos(-5.0, 5.0);
+    std::uniform_real_distribution<float> dist_radius(0.1, 1.0);
+    std::uniform_int_distribution<uint8_t> dist_color(0, 255);
+
+    std::vector<rr::components::Point3D> points3d(10);
+    std::generate(points3d.begin(), points3d.end(), [&] {
```
Comment (Wumpf) [inline docs/code-examples/point3d_random_v2.cpp:19]: doesn't compile on ci yet.

Ahhh C++ where you need to test each compiler individually. I bet my std lib impl leaked the necessary header in and the one on CI doesn't

File: docs/code-examples/disconnected_space_v2.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    auto rr_stream = rr::RecordingStream("disconnected_space");
+    rr_stream.connect("127.0.0.1:9876");
+
+    // These two points can be projected into the same space..
+    rr_stream.log(
+        "world/room1/point",
+        rr::archetypes::Points3D(rr::datatypes::Vec3D{0.0f, 0.0f, 0.0f})
+    );
+    rr_stream.log(
+        "world/room1/point",
```
Comment (teh-cmc) [inline docs/code-examples/disconnected_space_v2.cpp:17]: `world/room2/point`

File: docs/code-examples/disconnected_space_v2.cpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        rr::archetypes::Points3D(rr::datatypes::Vec3D{1.0f, 1.0f, 1.0f})
+    );
+
+    // ..but this one lives in a completely separate space!
+    rr_stream.log("world/room1/point", rr::archetypes::DisconnectedSpace(true));
```
Comment (teh-cmc) [inline docs/code-examples/disconnected_space_v2.cpp:22]: `world/wormhole`

File: docs/code-examples/disconnected_space_v2.cpp [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    rr_stream.log(
+        "world/room1/point",
```
Comment (teh-cmc) [inline docs/code-examples/disconnected_space_v2.cpp:24]: `world/wormhole/point`

File: docs/code-examples/point3d_random_v2.cpp [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        return rr::datatypes::Vec3D{dist_pos(gen), dist_pos(gen), dist_pos(gen)};
+    });
+    std::vector<rr::components::Color> colors(10);
+    std::generate(colors.begin(), colors.end(), [&] {
+        // TODO(andreas): provide a `rgb` factory method.
+        return (dist_color(gen) << 24) + (dist_color(gen) << 16) + (dist_color(gen) << 8);
```
Comment (teh-cmc) [inline docs/code-examples/point3d_random_v2.cpp:26]: This is logging colors with `0` alpha so it looks a bit weird in the demo right now:
image

**Review by teh-cmc — APPROVED** [review]
:+1:

File: docs/code-examples/point3d_random_v2.cpp (hunk 5 above)
Comment (Wumpf) [reply docs/code-examples/point3d_random_v2.cpp:26]: good catch. fixing

**Pull request merged**